### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by DariaSlayer**
- [DEMO LINK](https://DariaSlayer.github.io/layout_product-cards/)
- [TEST REPORT LINK](https://DariaSlayer.github.io/layout_product-cards/report/html_report/)

I would really appreciate some help to understand why can't I pass the test even if demo link looks legit

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -5,9 +5,46 @@
...
   <body>
-    <h1>Product cards</h1>
+    <main>
+      <div class="card" data-qa="card">
+        <img
+          class="card__picture"
+          src="../src/images/imac.jpeg"
+          alt="APPLE A1419 iMac 27inches Retina 5k Monoblock"
+        >
+        </img>
```
Comment (Andriy-Shymkiv) [inline src/index.html:20]: `img` is a self-closing tag

File: src/index.html [hunk 2]
```diff
@@ -5,9 +5,46 @@
...
+            <div class="stars__star"></div>
+            <div class="stars__star"></div>
+            <div class="stars__star"></div>
+            <div class="stars__star"></div>
+            <div class="stars__star"></div>
+          </div>
+          <div class="card__rating-text">Reviews: 5</div>
+        </div>
+        <div class="card__pricing">
+          <div class="card__pricing-text">Price:</div> <!-- if made into p or h the flex-end doesnt work -->
```
Comment (Andriy-Shymkiv) [inline src/index.html:36]: why not just `<p>` and `align-items: center;` ?

File: src/index.html [hunk 3]
```diff
@@ -5,9 +5,46 @@
...
+
+        <div class="card__rating">
+          <div class="stars stars--4">
+            <div class="stars__star"></div>
+            <div class="stars__star"></div>
+            <div class="stars__star"></div>
+            <div class="stars__star"></div>
+            <div class="stars__star"></div>
+          </div>
+          <div class="card__rating-text">Reviews: 5</div>
```
Comment (Andriy-Shymkiv) [inline src/index.html:33]: please use `<p>`, do not forget about SEO optimisation

File: src/index.html [hunk 4]
```diff
@@ -5,9 +5,46 @@
...
+        </div>
+        <div class="card__pricing">
+          <div class="card__pricing-text">Price:</div> <!-- if made into p or h the flex-end doesnt work -->
+
+          <div class="card__pricing-price">$2,199</div>
```
Comment (Andriy-Shymkiv) [inline src/index.html:38]: same here, use `p`

File: src/index.html [hunk 5]
```diff
@@ -5,9 +5,46 @@
...
+        </div>
+        <a
+          href="#"
```
Comment (Andriy-Shymkiv) [inline src/index.html:41]: add more meaningful href

File: src/styles/card.scss [hunk 6]
```diff
@@ -0,0 +1,86 @@
+
+.card {
+  display: flexbox;
```
Comment (Andriy-Shymkiv) [inline src/styles/card.scss:3]: `display: flexbox;`? is that a new display property or I missed something? 😄

File: src/styles/card.scss [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  box-sizing: border-box;
+  width: 200px;
+  padding-inline: 16px;
+  padding-top: 32px;
+  padding-bottom: 16px;
```
Comment (Andriy-Shymkiv) [inline src/styles/card.scss:8]: you can just - `padding: 32px 16px 16px`

File: src/styles/card.scss [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+  border-radius: 5px;
+  border: 1px solid #f3f3f3;
+  background: #fff;
```
Comment (Andriy-Shymkiv) [inline src/styles/card.scss:11]: I think background is white by default. isn't it?

File: src/styles/card.scss [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+  &__picture {
+    display: block;
+    width: 160px;
+    margin: 0 auto 40px;
+    object-position: center;
```
Comment (Andriy-Shymkiv) [inline src/styles/card.scss:17]: this property does not affect anything. what is this for?

File: src/styles/card.scss [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+  }
+
+  &__product-title {
+    font-size: $mainElementsSize;
+    font-weight: 500;
+    line-height: $mainLineHeight;
+    margin-block: 0 4px;
```
Comment (Andriy-Shymkiv) [inline src/styles/card.scss:24]: you can just `margin-bottom: 4px;` here and on other places below

File: src/styles/mixin.scss [hunk 11]
```diff
@@ -0,0 +1,5 @@
+@mixin space-between-content {
+  display: flex;
+  justify-content: space-between;
+  align-items: flex-end;
```
Comment (Andriy-Shymkiv) [inline src/styles/mixin.scss:4]: I think using `center` would be better in this case

**Review by Andriy-Shymkiv — CHANGES_REQUESTED** [review]
good job! left comments

File: src/index.html (hunk 2 above)
Comment (DariaSlayer) [reply src/index.html:36]: because I wanted to use mixing..... 🥺

File: src/styles/card.scss (hunk 6 above)
Comment (DariaSlayer) [reply src/styles/card.scss:3]: I'm somewhat an innovator myself 
hahaha
thanks and sorry

File: src/styles/card.scss (hunk 10 above)
Comment (DariaSlayer) [reply src/styles/card.scss:24]: then it adds extra space on top and I don't know why

**Comment by DariaSlayer** [comment]
I tried to fix the mistakes but switching it to p and h made blocks bigger and I'm not sure what to do with margins then. also couldn't switch margin block for margin bottom cause it gives extra margins on top for some reason. so I switched back to margin block to remove them

**Review by lerastarynets — CHANGES_REQUESTED** [review]
Pls try to pass all tests from backstop before asking for review again)

**Comment by DariaSlayer** [comment]
> Pls try to pass all tests from backstop before asking for review again)
> 

maybe if you looked at the demo link you'd see that everything looks good. and the only issue is test passing. so you could actually explain what the issue is then with it. then just writing such a comment

File: src/index.html [hunk 12]
```diff
@@ -5,9 +5,45 @@
...
+
+        <div class="card__rating">
+          <div class="stars stars--4">
+            <div class="stars__star"></div>
+            <div class="stars__star"></div>
+            <div class="stars__star"></div>
+            <div class="stars__star"></div>
+            <div class="stars__star"></div>
+          </div>
+          <p class="card__rating-text">Reviews: 5</p>
```
Comment (Esceype) [inline src/index.html:32]: You need to reset margins for p tag, some tags have their default setting, and if you work with designed mockup you need to reset all the tags, at future work you will have some reset.scss file or something like that. But now you need to do it by your own. So reset the margins for p.

File: src/styles/html.scss [hunk 13]
```diff
@@ -0,0 +1,6 @@
+html {
```
Comment (Esceype) [inline src/styles/html.scss:1]: Here you need to reset margins in body tag not in html.

File: src/styles/card.scss [hunk 14]
```diff
@@ -0,0 +1,82 @@
+
+.card {
+  display: flex;
+  box-sizing: border-box;
```
Comment (Esceype) [inline src/styles/card.scss:4]: Move this property to * tag cause this doesn`t work for button here

File: src/styles/card.scss [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+
+    &-price {
+      text-align: right;
+      font-size: 16px;
+      font-weight: $boldFontWeight;
+      line-height: $mainLineHeight;
+    }
+  }
+
+  &__button {
```
Comment (Esceype) [inline src/styles/card.scss:62]: Here add the static height of this element.

File: src/styles/card.scss [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+    width: 160px;
+    margin: 0 auto 40px;
+  }
+
+  &__product-title {
+    font-size: $mainElementsSize;
+    font-weight: 500;
+    line-height: $mainLineHeight;
+
+    @include block-distance;
```
Comment (Esceype) [inline src/styles/card.scss:22]: Reset standart h1 styles and then change margin-block to margin-bottom. There trouble can be in tests and they are not using margin-block in logic, as a variant. So try it

File: src/styles/card.scss [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+  }
+
+  &__product-code {
+    font-size: $secondaryElementsSize;
+    line-height: $secondaryLineHeight;
+    color: $secondaryColour;
+
+    @include block-distance;
```
Comment (Esceype) [inline src/styles/card.scss:30]: This will be wrong margin, just add margin bottom as in mockup

File: src/styles/card.scss [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+  display: flex;
+  box-sizing: border-box;
+  flex-direction: column;
+  flex-wrap: wrap;
+  width: 200px;
+  padding: 32px 16px 16px;
+  border-radius: 5px;
+  border: 1px solid #f3f3f3;
+
+  &__picture {
```
Comment (Esceype) [inline src/styles/card.scss:12]: Add static height here as in design

File: src/styles/card.scss [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+
+  &__product-code {
+    font-size: $secondaryElementsSize;
+    line-height: $secondaryLineHeight;
+    color: $secondaryColour;
+
+    @include block-distance;
+  }
+
+  &__rating {
```
Comment (Esceype) [inline src/styles/card.scss:33]: Here you also need to add margin bottom

File: src/styles/card.scss [hunk 20]
```diff
@@ -0,0 +1,82 @@
...
+    @include space-between-content;
+
+    &-text {
+      text-align: right;
+      font-size: $secondaryElementsSize;
+      line-height: $secondaryLineHeight;
+    }
+  }
+
+  &__pricing {
```
Comment (Esceype) [inline src/styles/card.scss:43]: Also you need to add margin bottom here

File: src/styles/card.scss [hunk 21]
```diff
@@ -0,0 +1,82 @@
+
+.card {
+  display: flex;
+  box-sizing: border-box;
+  flex-direction: column;
+  flex-wrap: wrap;
```
Comment (Esceype) [inline src/styles/card.scss:6]: Here you don`t need it

File: src/styles/card.scss [hunk 22]
```diff
@@ -0,0 +1,82 @@
...
+    }
+  }
+
+  &__button {
+    text-decoration: none;
+    text-align: center;
+    color: #fff;
+    background-color: $buttonColour;
+    border-radius: 5px;
+    display: flex;
```
Comment (Esceype) [inline src/styles/card.scss:68]: After you add the static height then remove flex-box and add width:100%

File: src/styles/card.scss [hunk 23]
```diff
@@ -0,0 +1,82 @@
...
+    font-size: 14px;
+    font-weight: $boldFontWeight;
+    text-transform: uppercase;
+    line-height: 16px;
+    padding-block: 12px;
+    padding-inline: 69px;
```
Comment (Esceype) [inline src/styles/card.scss:74]:
```suggestion
    padding: 12px 0;
```
Make it after all the previous comments

**Review by Esceype — CHANGES_REQUESTED** [review]
Great job!!! In general, I think what your tests don't go through is that you have rather non-standard logic, you use mixins a lot, maybe a bit too often at times, which is not very good. You should correct everything that I described in the comments. The main problem with the reset is also that the tests work with new CSS properties, here you often use padding-inline, or padding-block or margin-block, try to rewrite it using comments

File: src/styles/card.scss (hunk 21 above)
Comment (DariaSlayer) [reply src/styles/card.scss:6]: when I tried to move it to the body it didn't work. so I had to bring it back

**Comment by DariaSlayer** [comment]
I made some changes following @Esceype suggestion. but after following all the rules it made it look weird and it all flew away. so I followed most of them except a few. for it to look exactly like design and pass the tests (finally)

File: src/index.html [hunk 24]
```diff
@@ -5,9 +5,45 @@
...
+        </div>
```
Comment (denys-danyliuk) [inline src/index.html:33]:
```suggestion
        <div class="card__rating">
          <div class="stars stars--4">
            <div class="stars__star"></div>
            <div class="stars__star"></div>
            <div class="stars__star"></div>
            <div class="stars__star"></div>
            <div class="stars__star"></div>
          </div>
          
          <p class="card__rating-text">Reviews: 5</p>
        </div>
```

File: src/index.html [hunk 25]
```diff
@@ -5,9 +5,45 @@
...
+        <div class="card__pricing">
+          <p class="card__pricing-text">Price:</p> <!-- if made into p or h the flex-end doesnt work -->
+
+          <p class="card__pricing-price">$2,199</p>
+        </div>
+        <a
```
Comment (denys-danyliuk) [inline src/index.html:39]:
```suggestion
        <div class="card__pricing">
          <p class="card__pricing-text">Price:</p> <!-- if made into p or h the flex-end doesnt work -->
          <p class="card__pricing-price">$2,199</p>
        </div>
        
        <a
```

**Review by denys-danyliuk — APPROVED** [review]

**Pull request closed without merging**